src/core.rs

```diff
@@ -388,7 +388,7 @@ fn installed(config: &Config) -> Result<PackageIds> {
                 $(
                     $lower_backend:
                         if config.enabled_backends.contains(&AnyBackend::$upper_backend) {
-                            $upper_backend::query(config)?.keys().cloned().collect()
+                            $upper_backend::query(config.as_ref())?.keys().cloned().collect()
```

Comment:
Instead of `.as_ref()` could we not do `.$lower_backend`? Then we don't need the `?` as it would be simply struct field access.

Reply:
It works if we decide that all backends have a config and it is part of the main Config struct. See other comment

Note however the ? is for the query() result and was already there